File: solutions_6404600001200128_1/Python/chrisx/2015-1A-A.py
def parse(input_file, output_file):
    with open(input_file) as f:
        n_cases = int(f.readline().split()[0])
        data = dict()
        lengths = [0] * n_cases
        for i in range(n_cases):
            lengths[i] = int(f.readline().split()[0])
            data[i] = [int(x) for x in f.readline().split()]
    f = open(output_file, 'w')
    for i in range(n_cases):
        ans1 = solve1(lengths[i], data[i])
        ans2 = solve2(lengths[i], data[i])
        line = "Case #"+str(i+1)+": "+str(ans1)+" "+str(ans2)
        print(line)
        f.write(line+'\n')
    return


def solve1(length, seq):
    pre = seq[0]
    sum = 0
    for i in seq[1:]:
        #seqi = seq[i]
        if i<pre:
            sum += pre-i
        pre = i
    return sum

def solve2(length, seq):
    diff = [y-x for (x,y) in zip(seq[1:], seq[:-1])]
    uniform = [max(diff)] * (length-1)
    tosum = list(map(min, zip(uniform, seq[:-1])))
    return sum(tosum)


input_file = "./2015-1A/test.txt"
input_file = "./2015-1A/A-small-attempt0.in"
input_file = "./2015-1A/A-large.in"
output_file = "./2015-1A/A-large.out"
parse(input_file, output_file)

File: solutions_6404600001200128_0/C++/baixingdev/20151AA.cpp
#include<cstdio>
#include<algorithm>
using namespace std;

int T, _;
int n, v[1111];
int ans1, ans2, rate;

int main() {
	scanf("%d", &T);
	for(int _=1; _<=T; _++) {
		scanf("%d", &n);
		for(int i=0;i<n;i++)
			scanf("%d", &v[i]);
		ans1 = ans2 = 0;
		rate = 0;
		for(int i=1;i<n;i++) {
			if(v[i] < v[i-1]) ans1 += v[i-1] - v[i];
			
			rate = max(rate, v[i-1] - v[i]);
		}
		for(int i=0;i<n-1;i++){
			ans2 += min(rate, v[i]);
		}
		printf("Case #%d: %d %d\n", _, ans1, ans2);
	}

	return 0;
}